src/c4/ifpstream.cc

```diff
@@ -0,0 +1,71 @@
+//----------------------------------*-C++-*-----------------------------------//
+/*!
+ * \file   c4/ifpstream.cc
+ * \author Mathew Cleveland
+ * \date   Mon Jun 25 11:36:43 MDT 2018
+ * \brief  Define methods of class ifpstream
+ * \note   Copyright (C) 2018-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//----------------------------------------------------------------------------//
+
+#include "ifpstream.hh"
+#include "C4_Functions.hh"
+
+namespace rtt_c4 {
+using namespace std;
+
+//----------------------------------------------------------------------------//
+/*! Create an ifpstream to for a parallel file read.
+ *
+ * Create an ifpstream that reads and brocasts input file streams to all ranks
+ *
+ * \param[in] filename Name of the file to which is being read
+ * \param[in] mode File read mode (ascii/binary)-- defaults to ascii
+ */
+ifpstream::ifpstream(std::string const &filename,
+                     ios_base::openmode const mode) {
+  mode_ = mode;
+  if (rtt_c4::node() == 0) {
+    in_.open(filename, mode);
+  }
+}
+
+//----------------------------------------------------------------------------//
+/*! Fill parallel buffers with data from the input file
+ *
+ * \param[in] local process buffer size
+ */
+void ifpstream::fill_buffers(unsigned const buffer_size) {
```

Comment:
Is the idea that the calling code should have some (maybe problem-dependent) notion of the buffer_size? Is there some default we should set in the function definition?

Reply:
That is exactly right. It is problem dependent and therefor no sensible default can be set. The assumption is that who every wrote the file knows how they want to read it back in a parallel sense. Basically the originator (application specific) of the file should now what type of parallel stride the file should be read with.